File: implementation/scipamato/scipamato-public-entity/src/test/java/ch/difty/scipamato/publ/entity/StudyDesignCodeTest.java
package ch.difty.scipamato.publ.entity;

import static ch.difty.scipamato.publ.entity.StudyDesignCode.*;
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.extractProperty;

import org.junit.jupiter.api.Test;

class StudyDesignCodeTest {

    @Test
    void hasAllValues() {
        assertThat(values()).containsExactly(EXPERIMENTAL, EPIDEMIOLOGICAL, OVERVIEW_METHODOLOGY);
    }

    @Test
    void assertIds() {
        assertThat(extractProperty("id").from(values())).containsExactly((short) 1, (short) 2, (short) 3);
    }

    @Test
    void of_withExistingId() {
        assertThat(StudyDesignCode.of((short) 1)).hasValue(EXPERIMENTAL);
    }

    @Test
    void of_withNotExistingId_returnsEmptyOptional() {
        assertThat(StudyDesignCode.of((short) 0)).isEmpty();
    }
}

File: implementation/scipamato/public/entity/src/test/kotlin/ch/difty/scipamato/publ/entity/StudyDesignCodeTest.kt
package ch.difty.scipamato.publ.entity

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class StudyDesignCodeTest {

    @Test
    fun hasAllValues() {
        assertThat(StudyDesignCode.values()).containsExactly(
            StudyDesignCode.EXPERIMENTAL,
            StudyDesignCode.EPIDEMIOLOGICAL,
            StudyDesignCode.OVERVIEW_METHODOLOGY
        )
    }

    @Test
    fun assertIds() {
        assertThat(StudyDesignCode.values().map { it.id }).containsExactly(1.toShort(), 2.toShort(), 3.toShort()
        )
    }

    @Test
    fun of_withExistingId() {
        assertThat(StudyDesignCode.of(1.toShort())).hasValue(StudyDesignCode.EXPERIMENTAL)
    }

    @Test
    fun of_withNotExistingId_returnsEmptyOptional() {
        assertThat(StudyDesignCode.of(0.toShort())).isEmpty
    }
}
